gsdk/src/signer/calls.rs

```diff
@@ -55,7 +55,8 @@ use subxt::{
 type TxProgressT = TxProgress<GearConfig, OnlineClient<GearConfig>>;
 type EventsResult = Result<ExtrinsicEvents<GearConfig>, Error>;
 
-const ERRORS_REQUIRE_RETRYING: [&str; 2] = ["Connection reset by peer", "Connection refused"];
+// TODO: Somehow match error outside of async recursion in `sign_and_submit_then_watch`
```

Comment:
For me this issue should be solved within the PR.
Consider avoid async recursion at all by introducing some loop maybe, or by throwing waker for async runtime and introducing custom poll result?

Reply:
We need to completely remove retries because this is a userspace issue. You need to establish a connection to the rpc node before submitting tx.

Reply:
CI works